storage/range_command.go

```diff
@@ -1084,7 +1091,13 @@ func (r *Range) mergeTrigger(batch engine.Engine, merge *proto.MergeTrigger) err
 	// and not worth the extra logic and potential for error.
 	r.tsCache.Clear(r.rm.Clock())
 
-	return r.rm.MergeRange(r, merge.UpdatedDesc.EndKey, merge.SubsumedRaftID)
+	batch.Defer(func() {
```

Comment:
Also, doesn't everything that mutates `r` above need to be done in the defer (and Fatal when something goes wrong or somehow recover the previous state)?